known issue for events in 1.19 and 1.20 (#849)
hashicorp/web-unified-docs: content/vault/v1.19.x/content/docs/updates/important-changes.mdx

When two or more concurrent operations occur, Vault tries to reuse an idle
connection from the pool and the request fails due to session timeout in the
Snowflake database.

### Missed events with multiple event clients ((#missed-events)) <EnterpriseAlert inline="true" />

| Change       | Affected version       | Fixed version
| ------------ | ---------------------- | --------------------
| Known issue  | 1.20.x+ent, 1.19.x+ent | None

> Users may miss events if two or more event clients subscribe to the same
> performance standby node in a cluster, and each client specifies the same
> namespace and event type filters. If one client disconnects, the other(s) will
> no longer receive events.

Reviewer: ```suggestion
Users may miss events when multiple clients subscribe to the same performance
standby node in a cluster with the same namespace and event type filters because
one client disconnecting unsubscribes the remaining clients who no longer
receive events.
```
Style correction: write in active voice

#### Recommendation

> If there are multiple event subscribers with the same namespace and event type
> filters, spread them out among the nodes of the Vault cluster, or only subscribe
> to events on the active node of the cluster.

Reviewer: ```suggestion
If you have multiple event subscribers with the same namespace and event type
filters you have two options:

1. Spread them out among the nodes of the Vault cluster.
1. Only subscribe to events on the active node of the cluster.
```
